pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -322,4 +322,65 @@ async def build_sync(runtime: Runtime, version: str, assembly: str, publish: boo
         exclude_arches=exclude_arches,
         skip_multiarch_payload=skip_multiarch_payload
     )
-    await pipeline.run()
+    fail_count_name = f'{version}-assembly.{assembly}.count'
+
+    try:
+        await pipeline.run()
+
+        #  All good: reset fail count to 0
+        await redis.set_value(fail_count_name, 0)
+        runtime.logger.info('Fail count "%s" set to 0', fail_count_name)
+
+    except (RuntimeError, ChildProcessError):
```

Comment:
I think the redis setter should not be in the `try` block.
Because a failure to set the counter should be handled differently than a failure to run the pipeline. 
Also: How should the error / job result look like when the pipeline succeeds but setting count to `0` fails?

Reply:
`(RuntimeError, ChildProcessError)` will not catch the case where we fail to update the counter. This means, the job will break (and we'll know, which is good). Since this is an edge case, I think it's fine to leave the exception unhandled. It's something we want to be notified about ASAP